File: arthr.Models/Core/Staff.cs
namespace arthr.Models.Core
{
    #region Usings

    using System.Collections.Generic;
    using Notes;
    using Utils.Attributes;

    #endregion

    [DbEntity]
    public class Staff : Person
    {
        #region Properties

        public int StaffId { get; set; }

        public virtual ICollection<Note> Notes { get; set; }

        public virtual ICollection<StaffOnProjects> StaffOnProjects { get; set; }

        public virtual ICollection<StaffOnTask> StaffOnTasks { get; set; }

        public virtual User User { get; set; }
        public int UserId { get; set; }
        #endregion
    }
}

File: arthr.Web/Scripts/Models/Staff.ts
import { Note } from './Note';
import { StaffOnProjects } from './StaffOnProjects';
import { StaffOnTask } from './StaffOnTask';
import { User } from './User';
import { Person } from './Person';

export class Staff extends Person {
    staffId: number;
    notes: Note[];
    staffOnProjects: StaffOnProjects[];
    staffOnTasks: StaffOnTask[];
    user: User;
    userId: number;
}
/*
    Debug Info:
    Unknown Types: Note, StaffOnProjects, StaffOnTask, User
*/
